File: DataLayer/Entities/NeuralNetworkPerson.cs
using DataLayer.Extensions;

namespace DataLayer.Entities
{
    public class NeuralNetworkPerson : IJoinEntity<NeuralNetwork>, IJoinEntity<Person>
    {
        private NeuralNetwork navigation;
        private Person navigation1;
        public int NeuralNetworkId { get; set; }
        public int PersonId { get; set; }
        public NeuralNetwork NeuralNetwork { get; set; }
        public Person Person { get; set; }

        NeuralNetwork IJoinEntity<NeuralNetwork>.Navigation
        {
            get => navigation;
            set => navigation = value;
        }

        Person IJoinEntity<Person>.Navigation
        {
            get => navigation1;
            set => navigation1 = value;
        }
    }
}

File: dataLayer/entities/neural_network_person.py
from sqlalchemy import Column, String, Integer, Date, ForeignKey

from dataLayer.database_connection import Base


class NeuralNetworkPerson(Base):
    __tablename__ = 'NeuralNetworkPeople'

    neural_network_id = Column("NeuralNetworkId", Integer, ForeignKey("NeuralNetworks.Id"), primary_key=True)
    person_id = Column("PersonId", Integer, ForeignKey("Person.Id"), primary_key=True)
